src/bot.go

```diff
@@ -342,7 +342,19 @@ func (x *opBot) processJoinEvents(bot tgbotInterface, update tgbotapi.Update) {
 		tgbotapi.NewInlineKeyboardRow(buttonURL(T("visit_our_group_website"), osProgramadoresURL)),
 		tgbotapi.NewInlineKeyboardRow(buttonURL(T("read_the_rules"), osProgramadoresRulesURL)),
 	)
-	x.sendReplyWithMarkup(bot, update.Message.Chat.ID, update.Message.MessageID, fmt.Sprintf(T("welcome"), name), markup)
+	welcome, err := x.sendReplyWithMarkup(bot, update.Message.Chat.ID, update.Message.MessageID, fmt.Sprintf(T("welcome"), name), markup)
+	if err != nil {
+		log.Printf("Error sending welcomg message to user %s", name)
+		return
+	}
+
+	// Delete welcome message after a suitable timeout.
+	time.AfterFunc(30*time.Minute, func() {
```

Comment:
I am okay with this, but wouldn't it be better if instead of hard-coding the 30min here you would define a variable that we would be able to change by issuing an admin-level command?